packages/integration-tests/src/resource-creation/auth_resource_creator.ts

```diff
@@ -0,0 +1,369 @@
+import {
+  CognitoIdentityProviderClient,
+  CreateGroupCommand,
+  CreateGroupCommandInput,
+  CreateIdentityProviderCommand,
+  CreateIdentityProviderCommandInput,
+  CreateUserPoolClientCommand,
+  CreateUserPoolClientCommandInput,
+  CreateUserPoolCommand,
+  CreateUserPoolCommandInput,
+  CreateUserPoolDomainCommand,
+  CreateUserPoolDomainCommandInput,
+  DeleteGroupCommand,
+  DeleteIdentityProviderCommand,
+  DeleteUserPoolClientCommand,
+  DeleteUserPoolCommand,
+  DeleteUserPoolDomainCommand,
+} from '@aws-sdk/client-cognito-identity-provider';
+import {
+  CreateRoleCommand,
+  CreateRoleCommandInput,
+  DeleteRoleCommand,
+  IAMClient,
+} from '@aws-sdk/client-iam';
+import {
+  CognitoIdentityClient,
+  CreateIdentityPoolCommand,
+  CreateIdentityPoolCommandInput,
+  DeleteIdentityPoolCommand,
+  SetIdentityPoolRolesCommand,
+} from '@aws-sdk/client-cognito-identity';
+import { shortUuid } from '../short_uuid.js';
+import { e2eToolingClientConfig } from '../e2e_tooling_client_config.js';
+const RESOURCE_TAG_KEY = 'amplify-e2e-resource';
+const RESOURCE_TAG_VALUE = 'true';
+const TEST_RESOURCE_TAGS = {
+  [RESOURCE_TAG_KEY]: RESOURCE_TAG_VALUE,
+};
+const TEST_RESOURCE_TAGS_ARRAY = [
+  { Key: RESOURCE_TAG_KEY, Value: RESOURCE_TAG_VALUE },
+];
+/**
+ * Provides a way to create auth resources using aws sdk
+ */
+export class AuthResourceCreator {
+  private cleanup: { run: () => Promise<void>; arn?: string; id?: string }[] =
+    [];
+
+  /**
+   * Setup a new auth resource creator
+   * @param cognitoIdentityProviderClient client
+   * @param cognitoIdentityClient client
+   * @param iamClient client
+   */
+  constructor(
+    private cognitoIdentityProviderClient: CognitoIdentityProviderClient = new CognitoIdentityProviderClient(
+      e2eToolingClientConfig
+    ),
+    private cognitoIdentityClient: CognitoIdentityClient = new CognitoIdentityClient(
+      e2eToolingClientConfig
+    ),
+    private iamClient: IAMClient = new IAMClient(e2eToolingClientConfig),
+    private createResourceNameSuffix: () => string = shortUuid
+  ) {}
+
+  cleanupResources = async () => {
```

Comment:
updated to use prefix; also added waiter to check if stack is deleted (since roles can't be deleted due to attached policies until the stack policies are removed)